src/getMessageAuthor.ts

```diff
@@ -5,12 +5,45 @@ import {senderFromPayload} from './github/webhook'
 import {SlackClient} from './slack/SlackClient'
 import {MemberWithProfile, MessageAuthor} from './slack/types'
 
+export const GH_MERGE_QUEUE_BOT_USERNAME = 'github-merge-queue[bot]'
+
 export async function getMessageAuthor(
   octokit: OctokitClient,
   slack: SlackClient
 ): Promise<MessageAuthor | null> {
   startGroup('Getting message author')
 
+  const author = await fetchAuthor(octokit, slack)
+
+  try {
+    if (author && GH_MERGE_QUEUE_BOT_USERNAME === author.username) {
+      info(
+        'Author is GH Merge Queue Bot User. Fetching actual author via PR info.'
+      )
+      return await getSenderFromPRMerger(octokit)
```

Comment:
Would it be more appropriate to move this logic into the existing `getGitHubUser` method?

I think there could be several benefits:

* It would be ideal to leverage the existing `fetchAuthor` Slack matching algorithm such that `mergedBy` users are actually [at]mentioned.
* We probably don't need to attempt `getByUsername` in this merge queue scenario
* It would be nice to leverage the existing `fetchAuthor` error handling

Reply:
Ah yea good call. The main problem with this is if there is no slack user found it would fallback on the merge queue bot user again. Ideally we'd have it fallback on the PR merger even if they wouldn't be able to be @ mentioned. But should be possible to handle that.